sympy/core/intfunc.py

```diff
@@ -0,0 +1,558 @@
+import math
+from functools import lru_cache
+
+from .sympify import sympify
+from .singleton import S
+from sympy.external.gmpy import gcd as number_gcd, lcm as number_lcm, sqrt, iroot
+from sympy.utilities.misc import as_int
+
+from mpmath.libmp import bitcount
+
+
+def ndigits(n, base=10):
+    """Return the number of digits needed to express n in give base.
+
+    Examples
+    ========
+
+    >>> from sympy.ntheory.digits import ndigits
+    >>> ndigits(10)
+    2
+    >>> ndigits(10, 2)  # 1010 -> 4 digits
+    4
+    >>> ndigits(-100, 16)  # -64 -> 2 digits
+    2
+
+
+    Parameters
+    ==========
+
+    n: integer
+        The number whose digits are returned.
+
+    b: integer
+        The base in which digits are computed.
+
+    See Also
+    ========
+    ntheory.digits.digits, ntheory.digits.count_digits
+    """
+    b = as_int(base)
+    if b < 2:
+        raise ValueError("base must be integer greater than 1")
+    if b == 2:
+        return bitcount(n)
+    n = abs(as_int(n))
+    if n < b:
+        return 1
+    d = math.floor(math.log10(n) / math.log10(b))
```

Comment:
Why log10? I would expect if anything log2 would be the fastest since integers use a base 2 representation internally.

Reply:
base 2 special cased now; 10 makes base 10 better and doesn't matter for the other bases